ui/src/tasks/actions/index.ts

```diff
@@ -415,7 +462,7 @@ export const updateScript = () => async (dispatch, getState: GetStateFunc) => {
       updatedTask.every = null
     }
 
-    await client.tasks.update(task.id, updatedTask)
+    await patchTask({taskID: task.id, data: updatedTask})
```

Comment:
The UI is already explicitly setting `cron` and `every` to null values if the other value is set:

https://github.com/influxdata/influxdb/pull/16353/files#diff-3084610c031fd81910381dcc8507c9fcR457-R463